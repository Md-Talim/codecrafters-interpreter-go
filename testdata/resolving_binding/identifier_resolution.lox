// Test Case 1
fun global() {
  print "global";
}

{
  fun f() {
    global();
  }

  f();

  fun global() {
    print "local";
  }

  f();
}

// Test Case 2
var x = "global";

fun outer() {
  var x = "outer";

  fun middle() {
    fun inner() {
      print x;
    }

    inner();

    var x = "middle";

    inner();
  }

  middle();
}

outer();

// Test Case 3
var count = 0;

{
  fun makeCounter() {
    fun counter() {
      count = count + 1;
      print count;
    }
    return counter;
  }

  var counter1 = makeCounter();
  counter1();
  counter1();

  var count = 0;

  counter1();
}