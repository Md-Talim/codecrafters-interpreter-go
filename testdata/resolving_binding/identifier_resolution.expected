global
global
outer
outer
1
2
3